tests/api.c

```diff
@@ -4554,6 +4617,35 @@ static void test_wolfSSL_BIO(void)
     #endif
 }
 
+static void test_wolfSSL_BIO_puts(void)
+{
+    #if defined(OPENSSL_EXTRA)
+    char buffer[20];
+    BIO* bio1;
+    BIO* bio2;
+
+    printf(testingFmt, "wolfSSL_BIO_puts()");
+
+    /* Creating and testing type BIO_s_bio */
+    AssertNotNull(bio1 = BIO_new(BIO_s_bio()));
+    AssertNotNull(bio2 = BIO_new(BIO_s_bio()));
+
+    AssertIntEQ(BIO_set_write_buf_size(bio1, 20), SSL_SUCCESS);
+    AssertIntEQ(BIO_set_write_buf_size(bio2, 8),  SSL_SUCCESS);
+    AssertIntEQ(BIO_make_bio_pair(bio1, bio2),    SSL_SUCCESS);
+
+    buffer[0]='a';buffer[1]='b';buffer[2]='c';buffer[3]='\0';
+    AssertIntEQ(BIO_puts(bio1, buffer), 3);
```

Comment:
Is there a way we can verify that BIO_puts() worked as we expected, and wrote the data that we passed in correctly?